### PR TITLE
U2: Add foam.u2.View and foam.u2.DAOList

### PR DESCRIPTION
**Pull request by bshepherdson**

### CODE REVIEW

File: src/lib/u2/daos.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      name: 'rowFactory',
+      required: true,
+      adapt: function(old, nu) {
+        if ( typeof nu === 'function' ) return nu;
+
+        if ( typeof nu === 'string' ) {
+          return function(args, ctx) {
+            return ctx.lookup(nu).create(args, ctx);
+          };
+        } else {
```
Comment (kgrgreer) [inline src/lib/u2/daos.js:41]: No need for an 'else' if you've already returned, but I don't mind if this is how you prefer to code.

File: src/lib/u2/daos.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  properties: [
+    {
+      class: 'Function',
+      name: 'rowFactory',
+      required: true,
+      adapt: function(old, nu) {
+        if ( typeof nu === 'function' ) return nu;
+
+        if ( typeof nu === 'string' ) {
+          return function(args, ctx) {
```
Comment (kgrgreer) [inline src/lib/u2/daos.js:38]: Some documentation is required here.

File: src/lib/u2/daos.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            return ctx.lookup(nu).create(args, ctx);
+          };
+        } else {
+          return function(args, ctx) {
```
Comment (kgrgreer) [inline src/lib/u2/daos.js:42]: Document use-cases.

File: src/lib/u2/daos.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  ],
+
+
+  methods: [
+    function initE() {
+      // Just starts out empty.
+      this.cssClass(this.myCls());
+
+      // Kick off the select(), that will populate the view.
+      this.data.on.put.sub(this.onDAOPut.bind(this));
```
Comment (kgrgreer) [inline src/lib/u2/daos.js]: If you're going to bind these, why not just make them listeners?

File: src/lib/u2/daos.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+      this.reloadData();
+    },
+
+    function onDAOPut(obj) {
+      if ( this.rows_[obj.id] ) {
+        this.rows_[obj.id].data = obj;
+        return;
+      }
+
+      var ctx = this.__subContext__.createSubContext();
```
Comment (kgrgreer) [inline src/lib/u2/daos.js:82]: 'Y' looked much better.  Not a real review comment, just lamenting.

File: src/lib/u2/daos.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+      var child = this.rowFactory({ data: obj }, ctx);
+      child.on('click', function() {
+        this.rowClick.pub(obj);
```
Comment (kgrgreer) [inline src/lib/u2/daos.js]: Why not just this.rowClick.pub.bind(this.rowClick.pub, obj)

**Comment by bshepherdson** [comment]
All addressed.

**Pull request closed without merging**